ratis-client/src/main/java/org/apache/ratis/client/impl/DataStreamClientImpl.java

```diff
@@ -70,13 +70,13 @@ public DataStreamClientImpl(
     private long streamOffset = 0;
 
     public DataStreamOutputImpl(RaftGroupId groupId) {
-      this(groupId, RaftClientImpl.nextCallId());
+      this(new RaftClientRequest(clientId, raftServer.getId(), groupId, RaftClientImpl.nextCallId(),
+          RaftClientRequest.writeRequestType()));
     }
```

Comment:
Let's remove this constructor and move the new RaftClientRequest(..) to  stream(RaftGroupId gid).
```
  @Override
  public DataStreamOutputRpc stream(RaftGroupId gid) {
    return stream(new RaftClientRequest(clientId, raftServer.getId(), groupId, RaftClientImpl.nextCallId(),
        RaftClientRequest.writeRequestType()));
  }
```